service/worker/stream.go

```diff
@@ -121,14 +122,21 @@ func (ps *PushStream) Stop() error {
 
 func (ps *PushStream) processPushStreamResponse(ctx context.Context, subModel *subscription.Model, data metrov1.PullResponse) {
 	logger.Ctx(ctx).Infow("worker: response", "data", data)
+
 	for _, message := range data.ReceivedMessages {
 		logger.Ctx(ps.ctx).Infow("worker: publishing response data to subscription endpoint")
 		if message.AckId == "" {
 			continue
 		}
 
 		postData := bytes.NewBuffer(message.Message.Data)
-		resp, err := http.Post(subModel.PushEndpoint, "application/json", postData)
+		// TODO: move http client creation during push stream init
+		httpClient := &http.Client{
```

Comment:
httpclient can be cached. Also, make sure the transport options are optimised.

Reply:
Yup. Had left a TODO for that. Changed the impl. Moved HTTP init inside new push stream and re-using that client.